pkg/controller/product/3scale_reconciler.go

```diff
@@ -0,0 +1,95 @@
+package product
+
+import (
+	"fmt"
+
+	capabilitiesv1beta1 "github.com/3scale/3scale-operator/pkg/apis/capabilities/v1beta1"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+
+	threescaleapi "github.com/3scale/3scale-porta-go-client/client"
+	"github.com/go-logr/logr"
+)
+
+type ThreescaleReconciler struct {
+	*reconcilers.BaseReconciler
+	resource            *capabilitiesv1beta1.Product
+	productEntity       *helper.ProductEntity
+	backendRemoteIndex  *helper.BackendAPIRemoteIndex
+	threescaleAPIClient *threescaleapi.ThreeScaleClient
+	logger              logr.Logger
+}
+
+func NewThreescaleReconciler(b *reconcilers.BaseReconciler, resource *capabilitiesv1beta1.Product, threescaleAPIClient *threescaleapi.ThreeScaleClient, backendRemoteIndex *helper.BackendAPIRemoteIndex) *ThreescaleReconciler {
+	return &ThreescaleReconciler{
+		BaseReconciler:      b,
+		resource:            resource,
+		threescaleAPIClient: threescaleAPIClient,
+		backendRemoteIndex:  backendRemoteIndex,
+		logger:              b.Logger().WithValues("3scale Reconciler", resource.Name),
+	}
+}
+
+func (t *ThreescaleReconciler) Reconcile() (*helper.ProductEntity, error) {
+	productEntity, err := t.reconcile3scaleProduct()
+	if err != nil {
+		return nil, err
+	}
+	t.productEntity = productEntity
+
+	taskRunner := helper.NewTaskRunner(nil, t.logger)
+	taskRunner.AddTask("SyncProduct", t.syncProduct)
```

Comment:
Why is syncProduct separated from reconcile3scaleProduct? other syncers contain creation/update/delete

Reply:
It is a good question. There is no good answer. 

`reconcile3scaleProduct` is kind of "CreateSimpleOnlyIfNotExist" method.
`syncProduct` reconciliates all attributes provided it already exists.

All tasks assume the product already exists, and they can access `productEntity` object to manipulate it. Add metrics, mappingrules, etc...

The full reconciliation of the product and attributes (not sub resources like metrics and so on) could be done in `reconcile3scaleProduct`

Reply:
:+1: